pkg/kubelet/cm/memorymanager/memory_manager.go

```diff
@@ -306,35 +308,90 @@ func (m *manager) policyRemoveContainerByRef(podUID string, containerName string
 	return err
 }
 
-func validatePreReservedMemory(nodeAllocatableReservation v1.ResourceList, preReservedMemory map[int]map[v1.ResourceName]resource.Quantity) error {
-	// TODO: this will check equality of total reserved memory by node allocatable feature and total pre-reserved memory
+func getTotalMemoryTypeReserved(preReservedMemory map[int]map[v1.ResourceName]resource.Quantity) map[v1.ResourceName]resource.Quantity {
+	totalMemoryType := map[v1.ResourceName]resource.Quantity{}
 
-	return nil
+	for _, node := range preReservedMemory {
+		for memType, memVal := range node {
+			if totalMem, exists := totalMemoryType[memType]; exists {
+				memVal.Add(totalMem)
+			}
+			totalMemoryType[memType] = memVal
+		}
+	}
+
+	return totalMemoryType
 }
 
-func getReservedMemory(machineInfo *cadvisorapi.MachineInfo, nodeAllocatableReservation v1.ResourceList, preReservedMemory map[int]map[v1.ResourceName]resource.Quantity) (reservedMemory, error) {
-	// TODO: we should add new kubelet parameter, and to get reserved memory per NUMA node from it
-	// currently we use kube-reserved + system-reserved + eviction reserve for each NUMA node, that creates memory over-consumption
-	// and no reservation for huge pages
+func validatePreReservedMemory(nodeAllocatableReservation v1.ResourceList, preReservedMemory map[int]map[v1.ResourceName]resource.Quantity) error {
+	totalMemoryType := getTotalMemoryTypeReserved(preReservedMemory)
 
-	if err := validatePreReservedMemory(nodeAllocatableReservation, preReservedMemory); err != nil {
-		return nil, err
+	commonMemoryTypeSet := make(map[v1.ResourceName]bool)
+	for resourceType := range totalMemoryType {
+		if !(corev1helper.IsHugePageResourceName(resourceType) || resourceType == v1.ResourceMemory) {
+			continue
+		}
+		commonMemoryTypeSet[resourceType] = true
+	}
+	for resourceType := range nodeAllocatableReservation {
+		if !(corev1helper.IsHugePageResourceName(resourceType) || resourceType == v1.ResourceMemory) {
+			continue
+		}
+		commonMemoryTypeSet[resourceType] = true
 	}
 
-	reserved := reservedMemory{}
-	for _, node := range machineInfo.Topology {
-		memory := nodeAllocatableReservation[v1.ResourceMemory]
-		if memory.IsZero() {
-			break
+	for resourceType := range commonMemoryTypeSet {
+		if !(corev1helper.IsHugePageResourceName(resourceType) || resourceType == v1.ResourceMemory) {
```

Comment:
nit: I think you do not need this check here, because you already check it under previous iterations

Reply:
You are right, corrected.